src/SFML/Window/WindowImpl.cpp

```diff
@@ -212,13 +212,13 @@ Event WindowImpl::pollEvent()
 
 
 ////////////////////////////////////////////////////////////
-Event WindowImpl::popEvent()
+std::optional<Event> WindowImpl::popEvent()
 {
-    Event event;
+    std::optional<Event> event; // Use a single local variable for NRVO
 
     if (!m_events.empty())
     {
-        event = m_events.front();
+        event.emplace(m_events.front());
```

Comment:
What does `emplace` offer here over `operator=`?

Reply:
It makes it clear that `m_events.front()` returns an `sf::Event` and not a `std::optional<sf::Event>`. If I used `operator=`, it wouldn't be clear from this line of code whether `m_event` was a container of events or of optional events. `.emplace` makes it clear.